lib/ansible/plugins/inventory/generator.py

```diff
@@ -124,9 +125,12 @@ def parse(self, inventory, loader, path, cache=False):
 
         config = self._read_config_data(path)
 
+        extra_vars = load_extra_vars(loader)
+
         template_inputs = product(*config['layers'].values())
         for item in template_inputs:
             template_vars = dict()
+            template_vars.update(extra_vars)
```

Comment:
This breaks backwards compatibility, add a new option (see constructed, use_extra_vars, use_vars_plugins) . After that you should add a changelog and tests. https://docs.ansible.com/ansible/latest/reference_appendices/release_and_maintenance.html#generating-changelogs